### ISSUE TITLE
When class is set, also the id will get regocnized

### ISSUE THREAD

**Issue by JPeer264** [JPeer264/node-rcs-core#51]
```js
rcs.selectorLibrary.set('.test');

const idTestSelector = rcs.selectorLibrary.get('#test');

// should be true, instead it will get renamed
t.is(dotTestSelector, '#test');
```

**Comment by X-Ryl669** [comment]
Is that still happening? Because if it's the case, then I could probably fix this too in my changes.

**Comment by JPeer264** [comment]
Yes that's still an issue. I think I already wrote tests for it but made it to `.skip`. If you remove `.skip` they should pass :)

**Comment by X-Ryl669** [comment]
I'm going to split the selector member to `idSelector` and `classSelector` (and reciprocally for `compressedSelectors`). This should also double the number of possible mapping (since a mapped `a` for ids can also be used for classes).

**Comment by X-Ryl669** [comment]
Ok, while working on this, I think the solution above is dumb. 
If I understand correctly, the only reason the `selectorLibrary` is dealing with both *id* and *class* is because of the possible attribute selector.
So instead of mixing the concept here, I think it'll be better to have an (abstract) class called `AttributeSelectorLibrary` deriving from BaseLibrary and only implementing `setAttributeSelector` and `replaceAttributeSelector`.
 
Then I would create 2 new files `idSelectorLibrary.js` and `classSelectorLibrary.js` 
These would derive from `AttributeSelectorLibrary` (and indirectly from `BaseLibrary`) and thus have their own `values` & `compressedValues` mapping.

This will simplify the logic from many places since all the cases to test for `.` or `#` would be removed completely and the regex will be simpler. Yet, the replace's code will have to deal with an additional BaseLibrary instance (but it already has many for keyframes and css variables too).

If it's clear, please think about it and warn me if I'm misunderstanding something before I start intense code surgery, and let me know...

**Comment by JPeer264** [comment]
Ya I also thought a lot about this. I have no real clue what's the best solution here. I also thought about just extra parameters, such as `{ class: boolean; id: boolean }`. That would produce less code.

I think separating these into own files would be great (I am also not really worried about an additional `BaseLibrary` instance). But, I don't know how smart that is, as ids and classes could have the same name. I still want to rename both with the same minified name. Unless, there should be a new option `{ replaceIds: boolean; replaceClasses: boolean  }`, which will be easier with your approach.

**Comment by JPeer264** [comment]
I am happy about every implementation, if the code is still readable and easy to understand ;)

**Comment by X-Ryl669** [comment]
> I still want to rename both with the same minified name
Why ?
Currently, I'm seeing that the code is re-using the `nameGenerator` "global" instance, and I don't understand why.
It should not be an issue that, for example, CSS variables are renamed with their own `NameGenerator` different from the selector's generator (since `--a` is different than `.a` in CSS and so is `#a`, there is no possible confusion). This has the large advantage of giving much more "compression" power to the software: 

If you have a lot of **id**, **class**, some **keyframes** and some CSS **variables**, then with independent name generators, you'll get 4x more "small" value to compress into (since `a` for class might be different from the `a` for id and so on, you can now have 23 classes + 23 id + etc that will be mapped to a single letter equivalent).

**Comment by X-Ryl669** [comment]
Is `option.extend` really useful ?
Right now, the only place it's being used is in test code, so I wonder if I need to replicate its behavior. (And it's a pain to reimplement now)

**Comment by X-Ryl669** [comment]
Ok, even if the test code now pass, there is still an issue with `replace/string.js` that's stripping the selector's first char before fetching the replacement. I think this code needs rework too to only split on `' '` and `','` (and no more on the selector first char).

Typically, this currently fails:
```html
<style>
.list { }
</style>
<div class='list'></div>
<table id="list"></table>
<script>
var el = document.querySelector('#list');
</script>
```
The selector should not be replaced here, since no CSS rules refers to it.
Yet, in `replace/string`, it's stripped from its hash and the selectors library is searched for `list`. 
Since it's not found in IdSelector but found in ClassSelector, it's replaced by `a` then the hash is added and it gives `#a`.

If I remove this default behavior in SelectorsLibrary, then many cases with JS will fail.


I've another idea here which I think should work:
There no case where you'll have 2 words without a selector prefix that can match both Class and Id selector ("list some" is an invalid CSS selector and will never happen in `document.getElementById` nor `document.getElementsByClassName`).
Thus, if we have a single word, then we'll allow the loose search (with fallback from Id to Class)
If we have more than a single word, then the selector prefix **must** be present than then we can redirect to the right library (Id or Class).

Said differently, the `replace/string` code must be revamped completely.

**Comment by X-Ryl669** [comment]
Please have a look of last commit of #95

**Comment by JPeer264** [comment]
> Thus, if we have a single word, then we'll allow the loose search (with fallback from Id to Class)
If we have more than a single word, then the selector prefix must be present so we can redirect to the right library (Id or Class).

That's actually pretty clever and very accurate I'd say

**Comment by JPeer264** [comment]
Closed due to #95
